mlflow/tracing/processor/mlflow.py

```diff
@@ -47,6 +47,12 @@ def __init__(self, span_exporter: SpanExporter, client: Optional[MlflowClient] =
         self._client = client or MlflowClient()
         self._trace_manager = InMemoryTraceManager.get_instance()
 
+        # We issue a warning when a trace is created under the default experiment.
+        # We only want to issue it once, but de-duplication of warnings module does
+        # not work in notebooks (https://github.com/ipython/ipython/issues/11207).
+        # Therefore, we keep track of the warning issuance state manually.
```

Comment:
> de-duplication of warnings module does not work

is this because we

Reply:
Got it. Should we replace `_logger.warning` with `warnings.warn`?

Reply:
Yup that makes sense!

Reply:
Rephrased the comment to explicitly describe `warnings.warn` was tried but does not work. We use `_logger` in another line in the same file so I kept the `_logger` usage itself for consistency.